node-backend/app/controllers/v1/recognitionController.js

```diff
@@ -0,0 +1,326 @@
+                        data: info,
+                      });
+                    })
+                    .catch((err /*eslint-disable-line no-unused-vars*/) => {
+                      res.status(500).send({
+                        error: {
+                          message: "internal server error",
+                        },
+                      });
+                    });
+                } else {
+                  res.status(404).send({
+                    error: {
+                      message: "User not found in specified organition",
+                    },
+                  });
+                }
+              })
+              .catch((err /*eslint-disable-line no-unused-vars*/) => {
+                res.status(500).send({
+                  error: {
+                    message: "internal server error",
+                  },
+                });
+              });
+          } else {
+            res.status(404).send({
+              error: {
+                message: "core value not found with specified organisation",
+              },
+            });
+          }
+        })
+        .catch((err /*eslint-disable-line no-unused-vars*/) => {
+          res.status(500).send({
+            error: {
+              message: "internal server error",
+            },
+          });
+        });
+    })
+    .catch((err) => {
+      res.status(400).send({
+        error: utility.getFormattedErrorObj(
+          "invalid recognition",
+          "invalid recognition Data",
+          err.errors
+        ),
+      });
+    });
+};
+
+exports.findOne = /*eslint-disable-line node/exports-style*/ async (
+  req,
+  res
+) => {
+  const idSchema = yup.object().shape({
+    id: yup
+      .number()
+      .required({ id: "required" })
+      .typeError({ id: "should be number " }),
+  });
+
+  idSchema
+    .validate({ id: req.params.id }, { abortEarly: false })
+    .then(() => {
+      Recognitions.findByPk(req.params.id)
+        .then((data) => {
+          if (data == null /*eslint-disable-line no-eq-null*/) {
+            res.status(404).send({
+              error: {
+                message: "Recognition with specified id is not found",
+              },
+            });
+          } else {
+            res.status(200).send({
+              data: data,
+            });
+          }
+        })
+        .catch(() => {
+          res.status(500).send({
+            error: {
+              message: "internal server error ",
+            },
+          });
+        });
+    })
+    .catch((err) => {
+      res.status(400).send({
+        error: utility.getFormattedErrorObj(
+          "invalid recognition",
+          "invalid recognition Data",
+          err.errors
+        ),
+      });
+    });
+};
+
+const CreateFilterQuery = (filterData, tokenData) => {
+  const sqlQuery = "select * from recognitions where given_for";
+  if (
+    filterData.given_for == filterData.given_by &&
+    filterData.core_value_id == filterData.given_by
```

Comment:
how core_value_id can be compared with given_by and how given_for can be equal to given_by?

Reply:
i will add a one condition in this if (filterData.given_for == filterData.given_by &&
    filterData.core_value_id == filterData.given_by && filterData.given_for == undefined)